src/session/session_dhandle.c

```diff
@@ -428,11 +442,21 @@ __session_get_dhandle(WT_SESSION_IMPL *session, const char *uri, const char *che
 {
     WT_DATA_HANDLE_CACHE *dhandle_cache;
     WT_DECL_RET;
+    bool is_checkpoint;
 
-    __session_find_dhandle(session, uri, checkpoint, &dhandle_cache);
-    if (dhandle_cache != NULL) {
-        session->dhandle = dhandle_cache->dhandle;
-        return (0);
+    /*
+     * If we get called as part of the checkpoint, and the handle is the checkpoint to be created,
+     * it is guaranteed that the handle won't exist in either session or the shared dhandle list.
+     * Skip searching through the session dhandle list.
+     */
+    is_checkpoint = (WT_SESSION_IS_CHECKPOINT(session) && checkpoint != NULL &&
```

Comment:
You don't need the local variable here as this is the same conditional you used above (without a local variable). Or perhaps use a local boolean in both.